Compile error with OpenBSD and clang 19

**Describe the bug**
jq will not compile for OpenBSD under clang 19 due to -Werror=implicit-function-declaration being automatic.

Sample errors:

       > src/builtin.c:247:31: error: call to undeclared function 'lgamma_r'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]                                                               
       >   247 |   jv ret = JV_ARRAY(jv_number(lgamma_r(jv_number_value(input), &sign)));                                                                                                                                                     
       >       |                               ^                                                                                                                                                                                              
       > src/builtin.c:1341:10: error: call to undeclared function 'timegm'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]                                                                
       >  1341 |   return timegm(tm);                                                                                                                                                                                                         
       >       |          ^ 

**To Reproduce**
./configure && make

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment (please complete the following information):**
OpenBSD 7.5, clang 19

**Additional context**
Fixed with CFLAGS=-D_BSD_SOURCE=1. Autoconf detects these symbols in -lm but they are not visible in headers by default.

--- a/src/builtin.c
+++ b/src/builtin.c
@@ -5,6 +5,9 @@
 # define _XPG6
 # define __EXTENSIONS__
 #endif
+#ifdef __OpenBSD__
+# define _BSD_SOURCE
+#endif
 #include <sys/time.h>
 #include <stdlib.h>
 #include <stddef.h>

--- a/tests/optional.test
+++ b/tests/optional.test
@@ -12,6 +12,11 @@ last(range(365 * 67)|("1970-03-01T01:02:03Z"|strptime("%Y-%m-%dT%H:%M:%SZ")|mkti
 null
 [2037,1,11,1,2,3,3,41]
 
+# Regression test for #3276
+fromdate
+"2038-01-19T03:14:08Z"
+2147483648
+
 # %e is not available on mingw/WIN32
 strftime("%A, %B %e, %Y")
 1435677542.822351

